dynamic-tools/zap-extensions/zapwso2jiraplugin/ExtensionJiraIssueCreater.java

```diff
@@ -0,0 +1,199 @@
+public class ExtensionJiraIssueCreater extends ExtensionAdaptor {
+
+    JiraRestClient jira = new JiraRestClient();
+    UpdateJiraAttachments updateJiraAttachments = new UpdateJiraAttachments();
+
+    private static final ImageIcon ICON = new ImageIcon(
+            ExtensionJiraIssueCreater.class.getResource(IssueCreatorConstants.RESOURCE + "/cake.png"));
+    private JiraIssueCreaterAPI api = null;
+    private ZapMenuItem menuExample = null;
+    private AbstractPanel statusPanel = null;
+    private static final Logger log = Logger.getRootLogger();
+
+    /**
+     *
+     */
+    public ExtensionJiraIssueCreater() {
+        super();
+        initialize();
+    }
+
+    /**
+     * @param name
+     */
+    public ExtensionJiraIssueCreater(String name) {
+        super(name);
+    }
+
+    /**
+     * This method initializes this
+     */
+    private void initialize() {
+        this.setName(IssueCreatorConstants.NAME);
+    }
+
+    @Override public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+
+        this.api = new JiraIssueCreaterAPI(this);
+        API.getInstance().registerApiImplementor(api);
+        log.info("access to jira issue creator is optained");
+
+    }
+
+    @Override public String getAuthor() {
+        return Constant.messages.getString(IssueCreatorConstants.PREFIX + ".author");
+    }
+
+    @Override public String getDescription() {
+        return Constant.messages.getString(IssueCreatorConstants.PREFIX + ".desc");
+    }
+
+    @Override public URL getURL() {
+        try {
+            return new URL(Constant.ZAP_EXTENSIONS_PAGE);
+        } catch (MalformedURLException e) {
+            return null;
+        }
+    }
+
+    /**
+     * createJiraIssues will be called when API action is executed
+     *
+     * @param jiraBaseURL  is URL upto the jira Login window, in WSO2 it is upto <URL>/jira
+     * @param jiraUserName username of JIRA user
+     * @param jiraPassword Password of the JIRA user
+     * @param projectKey-  is defined under every Jira project, when browsing an issue it will be the last parameter without the issue number
+     * @param product      - You can define a product name, which will be used in the summary to Identify the Issue clearly
+     * @param issueLabel   This is a custom feild defined in wso2 jira
+     * @param filePath-    This is the file path where ZAP generates its report. This includes file name too.
+     * @return
+     */
+
+    public void createJiraIssues(String jiraBaseURL, String jiraUserName, String jiraPassword, String projectKey,
+            String asssignee, String product, String issueLabel, String filePath, String workspace,
+            String backupFolder) {
+
+        String issue;
+        String credentials = jiraUserName + ":" + jiraPassword;
+        String auth = new String(Base64.encodeBase64(credentials.getBytes()));
+        String BASE_URL = jiraBaseURL;
+
+        log.info("Backup Folder Path" + backupFolder);
```

Comment:
Do we need to have this as an INFO - i.e. to print each time? Does the backupFolder value change?